### DISCUSSION TITLE
Moving line with mouse button and get it's coordinates

### DISCUSSION THREAD

**Discussion by Protomolekule** [highfestiva/finplot#491] [General]
I'm trying to move a previously created line with:
```
marker1 = fplt.add_line ((start_date, 460), (end_date, 460),
               color='#ff0000', interactive=True, ax=ax)
```
but didn't find a way to access this line.  I've tried to use
`fplt.set_mouse_callback(mycallback, ax=ax, when='click')`

but this will called on every click except when click is on the line...

My target is to create different lines in the plot and read out its values. If I move a point, the whole line should be moved (redraw e.g.)
The closest thing I found was _How to delete previous line(s)? #358_ , but no description how get the line name and its values.
Maybe I stand on the line, please give me a hint.

**Comment by highfestiva** [comment]
You could override the creation of the `FinPolyLine` with one you create yourself. For all points on a poly line to move, you need to:
1. Create a class `MyLine` which inherits `FinPolyLine`.
2. Your `MyLine` implementation needs to override the `movePoint` function, where you move all points on the poly line.
3. Set `finplot._create_poly_line` to point to your own function where you instead create a new `MyLine`.

To get the position of the poly line, you can get all the segments of it, and in each segment you'll find a what called a "handle". From each handle you can get the position. See finplot's `__init__.py` at about line 706-759 or so for the `FinPolyLine` implementation.

**Reply by Protomolekule** [reply]
Thanks, I saw this example, but had problems to integrate. I came over to create a QMenu() via
`self.ax_a.scene ().sigMouseClicked.connect (self.handle_click)` 
But indeed I will check again to implement the FinPlotLine solution you mentioned. Is more comfortable ;-)
